features/steps/dataset_feature.go

```diff
@@ -0,0 +1,240 @@
+		AddCheckFunc: func(name string, checker healthcheck.Checker) error { return nil },
+		StartFunc:    func(ctx context.Context) {},
+		StopFunc:     func() {},
+	}, nil
+}
+
+func (f *DatasetFeature) DoGetHTTPServer(bindAddr string, router http.Handler) service.HTTPServer {
+	f.HTTPServer.Addr = bindAddr
+	f.HTTPServer.Handler = router
+	return f.HTTPServer
+}
+
+// DoGetMongoDB returns a MongoDB
+func (f *DatasetFeature) DoGetMongoDB(ctx context.Context, cfg *config.Configuration) (store.MongoDB, error) {
+	return f.MongoClient, nil
+}
+
+func (f *DatasetFeature) DoGetGraphDBOk(ctx context.Context) (store.GraphDB, service.Closer, error) {
+	return &storeMock.GraphDBMock{CloseFunc: funcClose}, &serviceMock.CloserMock{CloseFunc: funcClose}, nil
+}
+
+func (f *DatasetFeature) DoGetKafkaProducerOk(ctx context.Context, cfg *config.Configuration) (kafka.IProducer, error) {
+	return &kafkatest.IProducerMock{
+		ChannelsFunc: func() *kafka.ProducerChannels {
+			return &kafka.ProducerChannels{}
+		},
+		CloseFunc: funcClose,
+	}, nil
+}
+
+func (f *DatasetFeature) IHaveTheseDatasets(datasetsJson *godog.DocString) error {
+
+	datasets := []models.Dataset{}
+	m := f.MongoClient
+
+	err := json.Unmarshal([]byte(datasetsJson.Content), &datasets)
+	if err != nil {
+		return err
+	}
+	s := m.Session.Copy()
+	defer s.Close()
+
+	for _, datasetDoc := range datasets {
+		f.putDatasetInDatabase(s, datasetDoc)
+	}
+
+	return nil
+}
+
+func (f *DatasetFeature) putDatasetInDatabase(s *mgo.Session, datasetDoc models.Dataset) {
+	datasetID := datasetDoc.ID
+
+	datasetUp := models.DatasetUpdate{
+		ID:      datasetID,
+		Next:    &datasetDoc,
+		Current: &datasetDoc,
+	}
+
+	update := bson.M{
+		"$set": datasetUp,
+		"$setOnInsert": bson.M{
+			"last_updated": time.Now(),
+		},
+	}
+	_, err := s.DB(f.MongoClient.Database).C("datasets").UpsertId(datasetID, update)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (f *DatasetFeature) IAmNotIdentified() error {
+	f.FakeAuthService.NewHandler().Get("/identity").Reply(401)
+	return nil
+}
+
+func (f *DatasetFeature) IAmIdentifiedAs(username string) error {
+	f.FakeAuthService.NewHandler().Get("/identity").Reply(200).BodyString(`{ "identifier": "` + username + `"}`)
+	return nil
+}
+
+func (f *DatasetFeature) PrivateEndpointsAreEnabled() error {
+	f.Config.EnablePrivateEndpoints = true
+	return nil
+}
+
+func (f *DatasetFeature) TheDocumentInTheDatabaseForIdShouldBe(documentId string, documentJson *godog.DocString) error {
+	s := f.MongoClient.Session.Copy()
+	defer s.Close()
+
+	var expectedDataset models.Dataset
+
+	json.Unmarshal([]byte(documentJson.Content), &expectedDataset)
+
+	filterCursor := s.DB(f.MongoClient.Database).C("datasets").FindId(documentId)
+
+	var link models.DatasetUpdate
+
+	if err := filterCursor.One(&link); err != nil {
+		return err
+	}
+
+	assert.Equal(&f.ErrorFeature, documentId, link.ID)
+
+	document := link.Next
+	output, _ := json.MarshalIndent(document, "", "\t")
```

Comment:
As above

Reply:
Actually i assume this can be removed, output created for println below which can be removed